[Swift in WebKit] Work towards modularizing WebCore private headers (Part 6)

rr-codes opened the pull request:
#### 6edea7eed32f44bc63ec3684c91521a9a072faae
<pre>
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 6)
<a href="https://rdar.apple.com/159037620">rdar://159037620</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=297834">https://bugs.webkit.org/show_bug.cgi?id=297834</a>

Reviewed by Aditya Keerthi.

Fix some more module verification violations by adding missing headers and making private headers
project and vice versa.

* Source/WebCore/Modules/mediasession/MediaMetadata.h:
* Source/WebCore/Modules/url-pattern/URLPattern.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/LoadableClassicScript.h:
* Source/WebCore/editing/cocoa/NodeHTMLConverter.h:
* Source/WebCore/layout/FormattingState.h:
* Source/WebCore/layout/floats/FloatingContext.h:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/FormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingContext.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingState.h:
* Source/WebCore/layout/formattingContexts/block/tablewrapper/TableWrapperBlockFormattingContext.h:
* Source/WebCore/layout/formattingContexts/block/tablewrapper/TableWrapperBlockFormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.h:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingUtils.h:
* Source/WebCore/layout/formattingContexts/inline/InlineFormattingContext.h:
* Source/WebCore/layout/formattingContexts/inline/IntrinsicWidthHandler.h:
* Source/WebCore/layout/formattingContexts/inline/TextOnlySimpleLineBuilder.h:
* Source/WebCore/layout/formattingContexts/table/TableFormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/table/TableFormattingQuirks.h:
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.h:
* Source/WebCore/layout/integration/flex/LayoutIntegrationFlexLayout.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContentPainter.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.h:
* Source/WebCore/layout/layouttree/LayoutContainingBlockChainIterator.h:
* Source/WebCore/platform/cocoa/SearchPopupMenuCocoa.h:
* Source/WebCore/platform/graphics/ca/cocoa/PlatformDynamicRangeLimitCocoa.h:
* Source/WebCore/rendering/LineClampUpdater.h:
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
* Source/WebCore/workers/service/RouterCondition.h:

Canonical link: <a href="https://commits.webkit.org/299117@main">https://commits.webkit.org/299117@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a3640e0be18b2442b536b18a898cb4542dbfd5ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117880 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37558 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28190 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124024 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69913 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c7b67d5f-0150-4e78-bd34-32d11a781f13) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119758 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38250 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46140 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89452 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59069 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e684c4eb-7eab-44d1-88e2-0e62150c1761) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120832 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30457 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105697 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69948 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d984d36f-dd8d-429a-b295-9b9c96b6062b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29521 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23813 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67687 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99877 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23992 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127107 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44783 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33726 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98120 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45144 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101922 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97908 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24913 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43303 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21277 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41210 "Hash a3640e0b for PR 49822 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44655 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50329 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44115 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47460 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45804 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/mac/WebCoreFullScreenWarningView.h:
```diff
@@ -27,8 +27,12 @@
 #if PLATFORM(MAC) && ENABLE(FULLSCREEN_API)
 
+#import <AppKit/NSBox.h>
+#import <WebCore/PlatformExportMacros.h>
```
Why do we need `PlatformExportMacros.h` here but not in `WebCoreFullScreenPlaceholderView.h` above? Both use `WEBCORE_EXPORT`.

Is the import really necessary?

rr-codes replied:
The import is necessary, since otherwise the definition of `WEBCORE_EXPORT` is unknown and thus a module violation. The reason it's not in the other file yet is because the verifier is complaining about other files before it inevitably ends up complaining about that one.

pxlcoder replied:
Approved, with the expectation this will come in Part N 😛

rr-codes committed:
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 6)
rdar://159037620
https://bugs.webkit.org/show_bug.cgi?id=297834

Reviewed by Aditya Keerthi.

Fix some more module verification violations by adding missing headers and making private headers
project and vice versa.

* Source/WebCore/Modules/mediasession/MediaMetadata.h:
* Source/WebCore/Modules/url-pattern/URLPattern.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/LoadableClassicScript.h:
* Source/WebCore/editing/cocoa/NodeHTMLConverter.h:
* Source/WebCore/layout/FormattingState.h:
* Source/WebCore/layout/floats/FloatingContext.h:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/FormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingContext.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingState.h:
* Source/WebCore/layout/formattingContexts/block/tablewrapper/TableWrapperBlockFormattingContext.h:
* Source/WebCore/layout/formattingContexts/block/tablewrapper/TableWrapperBlockFormattingQuirks.h:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.h:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingUtils.h:
* Source/WebCore/layout/formattingContexts/inline/InlineFormattingContext.h:
* Source/WebCore/layout/formattingContexts/inline/IntrinsicWidthHandler.h:
* Source/WebCore/layout/formattingContexts/inline/TextOnlySimpleLineBuilder.h:
* Source/WebCore/layout/formattingContexts/table/TableFormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/table/TableFormattingQuirks.h:
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.h:
* Source/WebCore/layout/integration/flex/LayoutIntegrationFlexLayout.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContentPainter.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.h:
* Source/WebCore/layout/layouttree/LayoutContainingBlockChainIterator.h:
* Source/WebCore/platform/cocoa/SearchPopupMenuCocoa.h:
* Source/WebCore/platform/graphics/ca/cocoa/PlatformDynamicRangeLimitCocoa.h:
* Source/WebCore/rendering/LineClampUpdater.h:
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
* Source/WebCore/workers/service/RouterCondition.h:

Canonical link: https://commits.webkit.org/299117@main
